pkg/util/validation.go

```diff
@@ -18,19 +18,32 @@ package util
 
 import (
 	"regexp"
-	"strings"
 )
 
-// IsDNSLabel tests for a string that conforms to the definition of a label in
-// DNS (RFC 1123).
-func IsDNSLabel(value string) bool {
-	return IsDNS1123Label(value)
+const LabelValueFmt string = "[-A-Za-z0-9_.]*"
+
+var labelValueRegexp = regexp.MustCompile("^" + LabelValueFmt + "$")
+
+const LabelValueMaxLength int = 63
+
+func IsValidLabelValue(value string) bool {
+	return (len(value) <= LabelValueMaxLength && labelValueRegexp.MatchString(value))
 }
 
-// IsDNSSubdomain tests for a string that conforms to the definition of a
-// subdomain in DNS (RFC 1123).
-func IsDNSSubdomain(value string) bool {
-	return IsDNS1123Subdomain(value)
+// Annotation values are opaque.
+func IsValidAnnotationValue(value string) bool {
+	return true
+}
+
+const kubeToken string = "[A-Za-z0-9]"
```

Comment:
I figured out that we can be even move general, as those definitions will be used for label value format as well. I went with kubeChar and extendedKubeChar.